compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -553,6 +553,11 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
                     )?;
                 }
             }
+            // The intrinsic represents whether the value is known to the optimizer (LLVM).
+            // We're not doing any optimizations here, so there is no optimizer that could know the value.
+            // (We know the value here in the machine of course, but this is the runtime of that code,
+            // not the optimization stage.)
+            sym::is_val_statically_known => ecx.write_scalar(Scalar::from_bool(false), dest)?,
```

Comment:
There should be a const test invoking this intrinsic. Probably somewhere in `tests/ui/consts`.

Reply:
If I understand correctly, calling a constant function in a non-constant function isn't guaranteed to return `false` (be evaluated), right? Just that it has the same behavior at runtime and const eval? I believe this is the case but this is what I've noticed in the test.

Reply:
That's why I said a *const test* -- it should invoke the intrinsic from a const.

Currently this codepath is untested and that's not great.

Reply:
Yeah, I added a test for this.